estimator/simulator.py

```diff
@@ -102,13 +102,92 @@ def GSA(d, n, q, beta, xi=1, tau=1, dual=False):
     return r
 
 
+def ZGSA(d, n, q, beta, xi=1, tau=1, dual=False):
+    from math import lgamma
+    from .util import gh_constant, small_slope_t8
+    """
+    Reduced lattice Z-shape following the Geometric Series Assumption as specified in
+    NTRU fatrigue [DucWoe21]
+    :param d: Lattice dimension.
+    :param n: The number of `q` vectors is `d-n`.
+    :param q: Modulus `q`
+    :param beta: Block size β.
+    :param xi: Scaling factor ξ for identity part.
+    :param dual: ignored, since GSA is self-dual: applying the GSA to the dual is equivalent to
+           applying it to the primal.
+    :returns: Log Gram-Schmidt norms
```

Comment:
Would be good to have a doctest as an example on how to call this

Reply:
Happy to add, but none of the other Simulator functions have doctests.

Reply:
Yeah, let's improve the code base … good ol' scope creep :)